### PR TITLE
Fix workflow permission check logic

### PR DESCRIPTION
**Pull request by tonidero**
### Description
This improves the permission checking logic for running github actions. We also have it setup to only run actions by organization members, but this acts as an additional check.

### CODE REVIEW

**Comment by tonidero** [comment]
@RCGitBot please test

**Comment by tonidero** [comment]
@RCGitBot please test

**Comment by tonidero** [comment]
@RCGitBot please test

File: .github/workflows/trigger_all_tests.yml [hunk 1]
```diff
@@ -13,11 +13,13 @@ jobs:
```
Comment (JayShortway) [inline .github/workflows/trigger_all_tests.yml:13]: Maybe we can add another `if` (right above this line 😅) to make sure this doesn't work on forks? 

```yaml
if: github.repository == 'RevenueCat/purchases-ios'
```

Comment (vegaro) [reply .github/workflows/trigger_all_tests.yml:13]: I think that check could also be removed in the fork though? but yes, another extra check I guess

Comment (JayShortway) [reply .github/workflows/trigger_all_tests.yml:13]: Yea I think so too 🤔 But I see this being done, e.g. here: https://github.com/cashapp/multiplatform-paging/blob/main-3.3.0-alpha02/.github/workflows/release.yaml#L25

Comment (tonidero) [reply .github/workflows/trigger_all_tests.yml:13]: yeah... will add it since I think it shouldn't hurt in any case 👍

**Review by JayShortway — APPROVED** [review]
LGTM!

**Comment by tonidero** [comment]
@RCGitBot please test

**Comment by tonidero** [comment]
Will merge this for now, we can figure any other issues in followup PRs

File: .github/workflows/trigger_all_tests.yml [hunk 2]
```diff
@@ -9,15 +9,18 @@ jobs:
...
     steps:
       - name: Check membership in RevenueCat Org
+        env:
+          READ_ORG_GITHUB_TOKEN: ${{ secrets.READ_ORG_GITHUB_TOKEN }}
         id: verify
         # ensure that only RevenueCat members can trigger this
         run: |
-          RESPONSE=$(curl https://api.github.com/orgs/RevenueCat/members/${{ github.event.comment.user.login }})
-          if [[ "$RESPONSE" == *"Not Found"* ]]; then
+          RESPONSE=$(curl -s -o /dev/null --head -w "%{http_code}" -H "Authorization: Bearer $READ_ORG_GITHUB_TOKEN" https://api.github.com/orgs/RevenueCat/members/${{ github.event.comment.user.login }})
```
Comment (aboedo) [inline .github/workflows/trigger_all_tests.yml:22]: we should probably add a comment and maybe use `\` to split this up into lines for clarity

**Review by aboedo — COMMENTED** [review]
thanks for taking care of it!

**Comment by vegarotest** [comment]
@RCGitBot please test

**Pull request merged**